java/java.lsp.server/src/org/netbeans/modules/java/lsp/server/debugging/attach/AttachConfigurations.java

```diff
@@ -0,0 +1,216 @@
+                    String hostName = getArgumentOrDefault(defaultArguments.get("hostname"), "localhost");          // NOI18N
+                    String port = getArgumentOrDefault(defaultArguments.get("port"), "<port of the debuggee JVM>"); // NOI18N
+                    connector = new DebugConnector(NAME_ATTACH_SOCKET, type,
+                            Arrays.asList(SOCKET_ARG_HOST, SOCKET_ARG_PORT),
+                            Arrays.asList(hostName, port));
+                    break;
+                }
+                case CONNECTOR_SHMEM: {
+                    String name = getArgumentOrDefault(defaultArguments.get("name"), "<shared memory name>");       // NOI18N
+                    connector = new DebugConnector(NAME_ATTACH_SHMEM, type,
+                            Collections.singletonList(SHMEM_ARG_NAME),
+                            Collections.singletonList(name));
+                    break;
+                }
+                default: {
+                    List<String> names = new ArrayList<>();
+                    List<String> values = new ArrayList<>();
+                    for (Connector.Argument arg : defaultArguments.values()) {
+                        if (arg.mustSpecify()) {
+                            names.add(arg.name());
+                            String value = arg.value();
+                            if (value.isEmpty()) {
+                                value = "<" + arg.description()+ ">";   // NOI18N
+                            }
+                            values.add(value);
+                        }
+                    }
+                    connector = new DebugConnector(NAME_ATTACH_BY + connectorName, type,
+                            names, values);
+                }
+            }
+            connectors.add(connector);
+        }
+        connectors.sort((c1, c2) -> c1.getName().compareToIgnoreCase(c2.getName()));
+        return connectors;
+    }
+
+    private static String getArgumentOrDefault(Connector.Argument arg, String def) {
+        if (arg != null) {
+            String value = arg.value();
+            if (!value.isEmpty()) {
+                return value;
+            }
+        }
+        return def;
+    }
+
+    public static CompletableFuture<Object> findProcessAttachTo(NbCodeLanguageClient client) {
+        return CompletableFuture.supplyAsync(() -> {
+            return listProcessesToAttachTo(client);
+        }, RP).thenCompose(params -> client.showQuickPick(params)).thenApply(itemsList -> {
+            if (itemsList == null || itemsList.isEmpty()) {
+                return null;
+            } else {
+                return itemsList.get(0).getUserData();
+            }
+        });
+    }
+
+    private static void notifyNoProcessesError(NbCodeLanguageClient client) {
+        MessageParams params = new MessageParams();
+        params.setMessage("No debuggable JVM process found.\nPlease be sure to use `-agentlib:jdwp=transport=dt_socket,server=y` option.");
```

Comment:
NbBundle localization ? More general question to other reviewers: is the LSP server code expected to be l10n the same way as framework / IDE ?

Reply:
VSCode extension is not localized. I didn't mark this with NOI18N and I plan to start a discussion about localization on Monday.

Reply:
I've added localization to Java code.